### ISSUE TITLE
MSVC Float comparison to 0.0 not decompiled properly

### ISSUE THREAD

**Issue by Kerilk** [NationalSecurityAgency/ghidra#122]
**Describe the bug**
On 32 bit x86 architecture (and possibly 64bit), MSVC generates a specific pattern to test equality to zero of floating point numbers (see: https://stackoverflow.com/a/46772747). This pattern is incorrectly decompiled.

**To Reproduce**
decompile this pattern:
```
0f 57 c0            XORPS       XMM0,XMM0
0f 2e c8            UCOMISS     XMM1,XMM0
9f                  LAHF
f6 c4 44            TEST        AH,44h
7b 28               JNP         loc_equal_zero
```

The code is decompiled as:
`if(false) { ... }`

Once the dead code removal is deactivated.

**Expected behavior**
The expected behavior is:
`if ( value_in_XMM1 != 0.0 ) { ... }`

**Environment:**
 - OS: Windows 10
 - Version 9.0

**Comment by partoftheworlD** [comment]
Small and temporary solution, this is the wrong check, but fix decompiler output. The template for conditional jumps does not work correctly according to some instructions.
Should be added in line `3408`:
```
:JNP rel8       is vexMode=0 & byte=0x7B; rel8                                  { if (PF==1) goto rel8; }
:JNP rel16      is vexMode=0 & opsize=0 & byte=0x0F; byte=0x8B; rel16           { if (PF==1) goto rel16; }
:JNP rel32      is vexMode=0 & opsize=1 & byte=0x0F; byte=0x8B; rel32           { if (PF==1) goto rel32; }
```

before:
![image](https://user-images.githubusercontent.com/6013494/54920388-2b8b4b00-4f14-11e9-8b4e-2c447f932247.png)
after:
![image](https://user-images.githubusercontent.com/6013494/54920225-c2a3d300-4f13-11e9-8b01-da4dca6efc02.png)

**Comment by Kerilk** [comment]
Thanks for the feedback.

**Comment by partoftheworlD** [comment]
Sorry, JNP no needed to add, it's problem with  ucommis.

**Comment by partoftheworlD** [comment]
One more way to fix is remove `PF = nan(val1) || nan(val2);` in fucompe macro.

**Comment by caheckman** [comment]
Should be fixed in master now and in upcoming 9.2.  We've added full support for the parity flag.
